certificateshop-web/src/main/java/com/epam/esm/controller/CertificateTagController.java

```diff
@@ -0,0 +1,92 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.model.impl.CertificateTag;
+import com.epam.esm.model.impl.GiftCertificate;
+import com.epam.esm.service.CertificateService;
+import com.epam.esm.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+/**
+ * API to work with {@link CertificateTag}s of the GiftCertificatesShop.
+ */
+@RestController
+@RequestMapping("/tags")
+public class CertificateTagController {
+    private final TagService tagService;
+
+    /**
+     * Constructs the {@link CertificateTagController}.
+     * @param tagService is the service to inject.
+     */
+    @Autowired
+    public CertificateTagController(TagService tagService) {
+        this.tagService = tagService;
+    }
+
+    /**
+     * The method that realises the 'GET /tags' query.
+     *
+     * @return {@link List<CertificateTag>} - all the {@link CertificateTag} in the system.
+     */
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public List<CertificateTag> tags() {
+        return tagService.findAllCertificateTags();
+    }
+
+    /**
+     * The method that realises the 'GET /tags/{tagId}' query.
+     *
+     * @param tagId is the ID of the {@link CertificateTag} to find.
+     * @return {@link CertificateTag} with the tagId if such an id exists in the system.
+     */
+    @GetMapping(value = "/{tagId}")
+    @ResponseStatus(HttpStatus.OK)
+    public CertificateTag tag(@PathVariable("tagId") long tagId) {
+        return tagService.findCertificateTagById(tagId);
+    }
+
+    /**
+     * The method that realises the 'DELETE /tags/{tagId}' query.
+     *
+     * @param tagId is the ID of the {@link CertificateTag} to delete.
+     */
+    @DeleteMapping(value = "/{tagId}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```

Comment:
It is better to return status code 200 when tag deleted and status code 404 when nothing to delete